### ISSUE TITLE
[BUG] Processing mods after copy spins for ever

### ISSUE THREAD

**Issue by anothergene** [FSGModding/FSG_Mod_Assistant#263]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After resolving a version difference, the scanning mods happens.  It stalls at 100% for ever.  512/512 in my case.  Have to kill the program and start over.  The scanning seems to work fine in other places, like on launch and finish fairly quickly.  

### Expected Behavior

scanning actually finishes.  

### Steps To Reproduce

1.  update a mod
2. resolve version difference in a different folder
3. watch it spin for ever. 

### Environment

```markdown
- Mod Assistant Version:5.1.1
- OS:Win 11
- Virtualized: No
- Game Verson: xbox store
```

### Anything else?

Not sure where log ifle is located.  can add it if you can tell me where it is.

**Comment by anothergene** [comment]
found the log  entries in the app
 DEBUG | mod-f22185418a857c98d0c5959aaff16318 >> Adding mod FROM cache :: FS22_yara_N_sensor_ALS
2024-10-19 10:21:59 |   DEBUG | mod-b58c419ce4dfded93b076de897cca0cd >> Adding mod FROM cache :: FS22_yardProductionPack
2024-10-19 10:21:59 |   DEBUG | mod-477daadc2357befa41d89d544568fa9b >> Adding mod FROM cache :: FS22_TargetFillLevel
2024-10-19 10:21:59 |   DEBUG | mod-d485de2c8c5e618fd6526cc74fb64724 >> Adding mod FROM cache :: FS22_theWesternWilds
2024-10-19 10:22:00 |   DEBUG | worker-thread-2::82504 >>  -- Starting Worker Process : Thread # 2 -- 
2024-10-19 10:22:00 |   DEBUG | worker-thread-1::70024 >>  -- Starting Worker Process : Thread # 1 -- 
2024-10-19 10:22:00 |   DEBUG | worker-thread-3::81288 >>  -- Starting Worker Process : Thread # 3 -- 
2024-10-19 10:22:00 |   DEBUG | worker-thread-1::70024 >>  -- Exit Call, Work Finished --
2024-10-19 10:22:00 |   DEBUG | worker-thread-3::81288 >>  -- Exit Call, Work Finished --
2024-10-19 10:22:00 |    INFO | mod-136c1376408d1a1e4467e47dd40dea82 >> Adding Mod File: FS22_HeadlandManagement
2024-10-19 10:22:00 |   DEBUG | worker-thread-2::82504 >> Sent(got) modRecord :: col_e8a00120e6ec916d19ba20e85a1cd7a2--FS22_HeadlandManagement
2024-10-19 10:22:00 |    INFO | mod-136c1376408d1a1e4467e47dd40dea82 >> Adding mod to cache
2024-10-19 10:22:00 |   DEBUG | worker-thread-2::82504 >>  -- Exit Call, Work Finished --

**Comment by jtsage** [comment]
hmm.  race condition I bet.  I'll dig into this, thanks

**Comment by jtsage** [comment]
I think I have this fixed in 5.1.5, please let me know if the problem persists.

**Comment by anothergene** [comment]
Seem to be working now!  Thank you.

**Comment by anothergene** [comment]
Spoke too soon.... I thought it was working for me, but it's doing it again.  Maybe it's something I'm doing?

**Comment by anothergene** [comment]
I had to do a complete reinstall of FS22 for various reasons and it seems to be behaving now.  Will keep an eye on this week.

**Comment by jtsage** [comment]
yeah, I'm wondering if the actual scanner was somehow failing and not being caught - I'll experiment a bit with trying to force that to happen and see what I can find

**Comment by anothergene** [comment]
bit of an update.  It seems only to get suck spinning when I do a copy after a check version copy/resolution.  Any other time it appears to finish fairly quickly.

**Comment by jtsage** [comment]
thank you for continuing to run this down, that gives me a narrower place to look for the bug, appreciate it

**Comment by anothergene** [comment]
note:  happens if in 2022 or 2025 mode.

**Comment by jtsage** [comment]
yeah. this is actually related to #285, kind of.  either way, it appears to be fixed in the upcoming 6.x branch
